src/software/sensor_fusion/detection.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <vector>
+
+#include "software/new_geom/angle.h"
+#include "software/new_geom/point.h"
+#include "software/util/time/timestamp.h"
+
+/**
+ * A lightweight datatype used to input new data into the filter.
+ * We do this rather than taking the SSL_Detection data directly
+ * so we can make this module more generic and abstract away
+ * the protobuf for testing
+ */
+struct RobotDetection
+{
+    unsigned int id;
+    Point position;
+    Angle orientation;
+    double confidence;
+    Timestamp timestamp;
+};
+
+/**
+ * A lightweight datatype used to input new data into the filter.
+ * We do this rather than taking the SSL_DetectionBall directly
+ * so we can make this module more generic and abstract away
+ * the protobuf for testing
+ */
+struct BallDetection
+{
+    // The position of the ball detection on the field, in meters
+    Point position;
+    // The timestamp of the detection. This is the timestamp for when the camera frame
+    // containing the detection was captured
+    Timestamp timestamp;
+
+    bool operator<(const BallDetection &b) const
```

Comment:
wouldn't we want a function like this for all detection types? Would it make sense to have a base `Detection` class which `BallDetection` and `RobotDetection` extend off of?

Reply:
I don't think a base class that just contains Timestamp is worth it

Reply:
Fair enough, It seemed like the timestamp, position and comparison are common to all detections